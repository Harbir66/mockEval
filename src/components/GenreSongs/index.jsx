import React from 'react';
import './GenreSongs.css';
import PropTypes from 'prop-types';
import getAllGenres from '../../utils/common';
import GenreCard from '../GenreCard';
import AllSongs from '../AllSongs';

function GenreSongs({ songs }) {
  const genres = getAllGenres(songs);
  return (
    <div className="genre-songs">
      {genres.map((genre) => {
        return (
          <div key={Math.random()}>
            <GenreCard genre={genre.name} />
            <AllSongs
              songs={songs.filter((song) => song.genre.name === genre.name)}
            />
          </div>
        );
      })}
    </div>
  );
}

export default GenreSongs;

GenreSongs.propTypes = {
  songs: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      imageUrl: PropTypes.string.isRequired,
      artist: PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
      }),
      genre: PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
      }),
    })
  ).isRequired,
};
