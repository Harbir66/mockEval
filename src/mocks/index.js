export const mockSongs = {
  data: [
    {
      id: '1',
      name: 'song1',
      imageUrl:
        'https://i.scdn.co/image/ab67616d0000b27386a8ab515de4b7aef28cd631',
      artist: {
        id: '1',
        name: 'artist1',
      },
      genre: {
        id: '1',
        name: 'genre1',
      },
    },
    {
      id: '2',
      name: 'song2',
      imageUrl:
        'https://i.scdn.co/image/ab67616d0000b27386a8ab515de4b7aef28cd631',
      artist: {
        id: '2',
        name: 'artist2',
      },
      genre: {
        id: '2',
        name: 'genre2',
      },
    },
  ],
};

export const mockGenres = [
  {
    id: '1',
    name: 'Bollywood',
  },
  {
    id: '2',
    name: 'Pop',
  },
];
