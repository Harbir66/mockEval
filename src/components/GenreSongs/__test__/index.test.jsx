import React from 'react';
import { render } from '@testing-library/react';
import GenreSongs from '..';
import { mockSongs } from '../../../mocks/index';

describe('GenreSongs', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it('should render correctly', () => {
    const { asFragment } = render(
      <GenreSongs key={Math.random()} songs={mockSongs.data} />
    );
    expect(asFragment).toMatchSnapshot();
  });
});
