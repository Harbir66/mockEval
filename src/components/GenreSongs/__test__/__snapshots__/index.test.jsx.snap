// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreSongs should render correctly 1`] = `[Function]`;
